sympy/ntheory/generate.py

```diff
@@ -244,11 +247,35 @@ def primepi(n):
         prime : Return the nth prime
     """
     n = int(n)
-    if n < 2:
-        return 0
-    else:
-        return sieve.search(n)[0]
-
+    if n<2:
+		return 0	
+	lim = int(n**0.5) #lim = floor(sqrt(n))
```

Comment:
It needs to be larger. Here's an example I found. 

```
In [76]: floor(sqrt(1000000100000000000000000000000000))
Out[76]: 31622778182822583

In [77]: int(1000000100000000000000000000000000**0.5)
Out[77]: 31622778182822584
```

but if there's no way it can ever finish for a number as large as that then don't worry about it.